spec/sec-properties-of-the-regexp-prototype-object-patch.html

```diff
@@ -102,7 +102,7 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
       <emu-alg>
         1. Assert: Type(_S_) is String.
         1. Assert: _Input_ is a List of the code points of _S_ interpreted as a UTF-16 encoded string.
-        1. Assert: _e_ is an integer value &ge; 0 and &lt; the number of elements in _Input_.
+        1. Assert: _e_ is an integer value &ge; 0.
```

Comment:
it seems worth it to specify that it's either "zero if Input is empty, or < the number of elements in Input"?

Reply:
That case is handled by step 4:

> 4. Let eUTF be the smallest index into S that corresponds to the character at element e of Input. If e is greater than or equal to the number of elements in Input, then eUTF is the number of code units in S.

Reply:
I can't recall what I was thinking when I asked this in April, but re-looking now, I'm wondering if that second sentence should be removed entirely - when would `e` be greater than the number of elements in `Input`? a misperforming subclass?

Reply:
I have it here because it exists in the spec text that this code replaces: https://github.com/tc39/ecma262/blob/d5948f7bf2a445ccd0a14ac9b06e57dcee397ed7/spec.html#L32144. I'm not sure of the historical reasons for this requirement, but it's possible this is to handle malformed surrogate pairs in `u` mode.

Reply:
I think the spec would read clearer here if, before this assertion, it has something like “if Input is empty, return 0”, rather than relying on the very hand-wavy language in the next line.

Reply:
Is that what the second line is intended to mean though? The original spec text is somewhat confusing, to be fair:

> 14. If _fullUnicode_ is *true*, then
>     1. _e_ is an index into the _Input_ character list, derived from _S_, matched by _matcher_. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _Input_. If _e_ is greater than or equal to the number of elements in _Input_, then _eUTF_ is the number of code units in _S_.

Reply:
I’d even prefer adding an explicit step handling the empty string and leaving the second sentence as it is, tbh.

Reply:
Is there a reason to handle the empty string separately if that is already covered by the behavior of the second line?

Reply:
I don't think i'd insist on it, but to me it's much clearer to read and understand - the same way that all the number/math functions handle infinites, NaN, and zeroes before attempting to handle the more meaty logic.